apps/dashboard/app/views/batch_connect/session_contexts/_path_selector.html.erb

```diff
@@ -6,23 +6,25 @@ end
 
 <%= label attrib.id, "#{field_options[:label]}" %>
 <%= text_field :batch_connect_session_context, attrib.id, class: 'form-control' %>
-<button type="button" class="btn btn-primary mt-2" data-toggle="modal" data-target="#modalPathSelector" id='select-nav-path'>
+<button type="button" class="btn btn-primary mt-2" data-toggle="modal" data-target="#modal-path-selector" id='select-nav-path'>
   Select Path
 </button>
 <span id='path_selector_home_dir' class="d-none"><%= Dir.home %></span>
 <span id='path_selector_url' class="d-none"><%= files_path %></span>
+<span id='path_selector_options' class="d-none"><%= attrib.opts[:options].is_a?(Array) ? attrib.opts[:options].join(",") : '' %></span>
```

Comment:
I think the prefererd way to embed information is to use `data-` attributes. So we could use 

`data-dot-files="true"` in the HTML

and in the javascript it becomes as easy as.

```javascript
function showDotFiles() {
  $('#modal-path-selector').data('showHidden') === true;
}
```

You can use a helper ruby function in `app/helpers/batch_connect/session_contexts_helper.rb` itself to hold some logic. 

Beyond that, I think this will help with using localStorage too, which I'm not sure we should be using. We can read the settings directly from the HTML instead and maybe save it in a variable?  It seems like we do need to save the result because it's used over and over again, but I wonder what that'll do to uniqueness. I mean imagine a use case where there are 2 of these `path_selector` widgets on the same page and they both update the same localStorage items and potentially conflict with each other?

Reply:
Is there not a higher cost of reading from the div id every time as opposed to reading the div id once and setting the object property, then reading from it?